source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,198 @@
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Retrieve.java
+   :language: java
+   :dedent:
+   :start-after: begin findExample
+   :end-before: end findExample
+
+For more information how to build filters, see our :doc:`Filters Builders
+</fundamentals/builders/filters>` guide.
+
+The following shows the output of the query specified above:
+
+.. code-block:: json
+
+    { "_id": 2, "color": "green", "qty": 8 }
+    { "_id": 3, "color": "purple", "qty": 4 }
+
+After the paint store runs this query, they find that orders with
+``_id`` "2" and "3" contain more than three, but less than ten cans
+of paint.
+
+For a runnable ``find()`` example, see our :doc:`Find Multiple
+Documents </usage-examples/find>` page. 
+
+.. _retrieve-aggregate:
+
+Aggregate
+~~~~~~~~~
+
+Use the aggregate operation to perform the stages in an aggregation
+pipeline.  An aggregation pipeline is a multi-staged transformation that
+produces an aggregated result. 
+
+To perform an aggregate operation, call the ``aggregate()`` method on an
+instance of a ``MongoCollection``. This method accepts aggregation
+expressions to run in sequence. To perform transformations, you can
+define aggregation stages that specify how to match documents, rename
+fields and group values. For more information on aggregation, see our
+:doc:`Aggregation  </fundamentals/aggregation>` guide.
+
+Example
+```````
+
+The paint store would like to know which color is the most purchased
+(highest quantity sold).
+
+To address the scenario, the paint store needs to create an aggregation
+pipeline that:
+
+- Matches all the documents in the ``paint_order`` collection
+- Groups orders by colors 
+- Sums up the quantity field by color
+- Sorts the results in descending order by quantity
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Retrieve.java
+   :language: java
+   :dedent:
+   :start-after: begin aggregateExample
+   :end-before: end aggregateExample
+
+The following shows the output of the query specified above:
+
+.. code-block:: json
+
+    { "_id": "green", "qty": 19 }
+    { "_id": "purple", "qty": 14 }
+
+After the paint store runs the query, they find that "green" is the most
+purchased color.
+
+For more information on how to construct an aggregation pipeline, see
+the MongoDB server manual page on :manual:`Aggregation </aggregation>`.
+
+.. _retrieve-watch:
+
+Change Streams
+~~~~~~~~~~~~~~
+
+Open a change stream to monitor real-time changes to a collection.
+Change streams can subscribe to types of data changes specified and 
+immediately produce change events. For more information on change
+streams, see the MongoDB server manual page on :manual:`Change Streams
+</changeStreams>`. 
+
+To open a change stream, call the ``watch()`` method on an instance of a
+``MongoCollection``, ``MongoDatabase`` or ``MongoClient``. You can pass
+an aggregation pipeline to this method to notify you and react as soon as a
```

Comment:
A couple items:

1) I think react is something that describes part of the process of creating a notification so you'd probably want to list it first if you needed to include both "notify" and "react".
2) I don't think react adds any value since "notify you" is the only item that is describing the outcome. 

```suggestion
an aggregation pipeline to this method to notify you as soon as a
```